docs/CONTRIBUTING.md

```diff
@@ -1,21 +1,147 @@
 
-- All contributions to this project are licensed under the terms of the Apache License, Version 2.0.
+* All contributions to this project are licensed under the terms of the Apache License, Version 2.0.
+ 
   [Read `LICENSE.txt` for details](../LICENSE.txt).
 
-## Get started
+There are multiple ways to contribute to the web-features repository, such as:
 
-The project is in its early stages, so we’re still developing tools and processes to help improve and expand our work.
-To get started contributing, take a look the following:
+* Adding a missing feature to the repository, from scratch.
+* Adding a missing feature to the repository, by updating a feature that was already drafted.
+* Updating an existing feature, for example to change its name, description, support data, or other fields.
+* Reviewing a pull request that was submitted by someone else, to check if the feature is correctly authored.
 
-* Go to [feature guidelines](./README.md#feature-guidelines).
-* Go to [open issues](https://github.com/web-platform-dx/web-features/issues) to propose new feature groups or report a bug.
-* Go to [open PRs](https://github.com/web-platform-dx/web-features/pulls) to review incoming feature groups and updates.
+In any case, **thank you** for wanting to help. This document will guide you through the process of contributing to the web-features repository.
 
-Also consider joining [the WebDX Community Group](https://www.w3.org/community/webdx/).
+Also consider joining the [WebDX Community Group](https://www.w3.org/community/webdx/).
+
+## What makes a feature
+
+Think of features as individual components of the web platform that web developers can use to achieve specific tasks.
+
+As such, a feature has no specific size. Some features might cover single CSS properties while other features cover entire JavaScript APIs with multiple interfaces, methods, and properties. As long as a web developer would use this to achieve a specific goal, it's a feature. 
+
+For example, the `fetch` API is a feature, the `:has()` CSS pseudo-class function too, and so is the Web Audio API.
+
+### Feature file name and format
+
+Features in this repository are authored as **YAML files** that are stored in the [`features`](https://github.com/web-platform-dx/web-features/tree/main/features) directory. Each file corresponds to a single feature, and contains metadata about the feature, such as its name, description, specification, or support data.
+
+For example, the `fetch` API feature is described in the [`features/fetch.yml`](https://github.com/web-platform-dx/web-features/blob/main/features/fetch.yml) file.
+
+The name of the file also constitutes the unique ID of the feature. In the example above, the unique ID of the feature is `fetch`.
+
+Feature IDs must be unique as they are used in other projects to reference the feature. For example, the [browser-compat-data](https://github.com/mdn/browser-compat-data/) project references feature IDs in their compatibility data.
```

Comment:
This would be a good place for a table or definition list with all of our fields in the YAML files. A TODO comment is fine though, if you want to focus on the narrative bits for now.